professional_python_exercises_2_githubcli/api/github_api.py

```diff
@@ -0,0 +1,116 @@
+import sys
+from http import HTTPStatus
+
+import requests
+import typer
+from github import Github
+
+from .repo_star_count import RepoStarCount
+from .token import get_github_token
+from .user_details import UserDetails
+
+
+class GitHubAPI:
+    """Class as interface to the GitHub API"""
+
+    def __init__(self, token: str) -> None:
+        self.session = Github(token)
```

Comment:
Love the way, the session is once created with init and then available again at a later stage :)